src/input/mod.rs

```diff
@@ -152,22 +150,30 @@ impl<'a> InputFile<'a> {
 
 
     /// Reads image data into the given FrameBufferMut.
+    ///
+    /// The passed FrameBufferMut must match the image's resolution exactly, and
```

Comment:
On second thought, since this leads to panics we probably should document it on each method, even if it's a bit redundant.  And when we get around to it we can note the bigger principle in higher-level documentation as well.